arangod/RocksDBEngine/RocksDBMetadata.h

```diff
@@ -74,6 +77,28 @@ struct RocksDBMetadata final {
  public:
   RocksDBMetadata();
 
+#ifdef ARANGODB_ENABLE_FAILURE_TESTS
+  // marks document counts as tainted during testing
+  void setTainted() { _tainted = true; }
+#endif
+
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+  bool tainted() const noexcept {
+#ifdef ARANGODB_ENABLE_FAILURE_TESTS
+    // if we use failure tests, the document counts may have been intentionally
+    // corrupted. the tainted state is set by the failure points that corrupt
+    // the counters.
+    return _tainted;
+#else
+    // if we don't use failure tests, the document counts are never tainted.
+    return false;
+#endif
+  }
+#else
+  // non-maintainer mode...
+  static constexpr bool tainted() const noexcept { return false; }
+#endif
```

Comment:
This could be simplified:
```suggestion
#if defined(ARANGODB_ENABLE_FAILURE_TESTS) && defined(ARANGODB_ENABLE_MAINTAINER_MODE)
  // if we use failure tests, the document counts may have been intentionally
  // corrupted. the tainted state is set by the failure points that corrupt
  // the counters.
  bool tainted() const noexcept { return _tainted; }
#else
  // non-maintainer mode or failure tests not enabled.
  static constexpr bool tainted() const noexcept { return false; }
#endif
```